### PR TITLE
Eng 1507 add stack trace to failed object deletion in workflow deletion [BE/3]

### PR DESCRIPTION
**Pull request by eunice-chan**
**skipping integration tests because it is currently failing integration tests. This is due to the fact the SDK side expects a `succeeded` field but that gets changed to `exec_state` and I update the SDK expectation in the next PR (Component https://github.com/aqueducthq/aqueduct/pull/342)**
## Describe your changes and why you are making these changes
Add stack traces / error messages to the failed object deletions as suggested in https://github.com/aqueducthq/aqueduct/pull/290.

BE Component https://github.com/aqueducthq/aqueduct/pull/340
UI Component https://github.com/aqueducthq/aqueduct/pull/341
SDK Component https://github.com/aqueducthq/aqueduct/pull/342

## Related issue number (if any)
ENG 1370
ENG 1507

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [N/A] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/python/aqueduct_executor/operators/utils/utils.py [hunk 1]
```diff
@@ -267,7 +267,10 @@ def write_delete_saved_objects_results(
 ) -> None:
     results_str = json.dumps(
         {
-            integration: [result.__dict__ for result in results[integration]]
+            integration: [
+                {"name": result.name, "exec_state": json.loads(result.exec_state.json())}
```
Comment (eunice-chan) [inline src/python/aqueduct_executor/operators/utils/utils.py:271]: [class].json() gives me the json as a string which gets escaped when I do `json.dumps`. However, I cannot serialize it directly with `json.dumps` so I serialize with `.json()` then `json.loads()` to get the format that is serializable by `json.dumps`.

File: src/python/aqueduct_executor/operators/connectors/tabular/relational.py [hunk 2]
```diff
@@ -34,13 +42,18 @@ def delete(self, tables: List[str]) -> List[SavedObjectDelete]:
         metadata = MetaData()
         metadata.reflect(bind=self.engine)
         for table in tables:
+            exec_state = ExecutionState(user_logs=Logs())
             try:
                 sql_table = metadata.tables[table]
                 Base.metadata.drop_all(self.engine, [sql_table], checkfirst=True)
-            except:
-                results.append(SavedObjectDelete(name=table, succeeded=False))
+            except Exception as e:
```
Comment (likawind) [inline src/python/aqueduct_executor/operators/connectors/tabular/relational.py:49]: We can do this as a follow-up, consider ways to de-duplicate these code esp. for the exception handling part

**Review by likawind — APPROVED** [review]
LGTM! One comment for non-immediate clean-up but I think this is good to go!

**Review by likawind — APPROVED** [review]
Looks good! Just try to rebase to ensure there are no merge conflicts :)

**Pull request merged**